src/main/java/jenkins/branch/MultiBranchProject.java

```diff
@@ -538,9 +553,17 @@ public P getItem(String name) {
             if (item != null) {
                 return item;
             }
+            item = super.getItem(NameMangler.apply(decoded));
```

Comment:
If you do getItem("foo%2fbar") as a historical link, then `foo%2fbar` will be decoded into `foo/bar` and then we check if there is an item with the name `NameMangler.apply("foo/bar")`, if there is then that was the item the old link was trying to access